python/ray/serve/_private/http_state.py

```diff
@@ -122,13 +122,17 @@ def update_actor_details(self, **kwargs) -> None:
 
     def _health_check(self):
         """Perform periodic health checks."""
+        assert self._status in {HTTPProxyStatus.HEALTHY, HTTPProxyStatus.DRAINING}
+
         if self._health_check_obj_ref:
             finished, _ = ray.wait([self._health_check_obj_ref], timeout=0)
             if finished:
                 self._health_check_obj_ref = None
                 try:
                     ray.get(finished[0])
-                    self.try_update_status(HTTPProxyStatus.HEALTHY)
+                    # Call to reset _consecutive_health_check_failures
+                    # the status should be unchanged.
+                    self.try_update_status(self._status)
```

Comment:
This is the bug I found during test: if a node is draining, we shouldn't set the status back to healthy even if health check succeeds.